src/openrct2/scripting/ScEntity.hpp

```diff
@@ -158,6 +183,220 @@ namespace OpenRCT2::Scripting
+        }
+        void nauseaTarget_set(uint8_t value)
+        {
+            ThrowIfGameStateNotMutable();
+            auto peep = GetPeep();
+            if (peep != nullptr)
+            {
+                peep->nausea_target = value;
+            }
+        }
+
+        uint8_t hunger_get() const
+        {
+            auto peep = GetPeep();
+            return peep != nullptr ? peep->hunger : 0;
+        }
+        void hunger_set(uint8_t value)
+        {
+            ThrowIfGameStateNotMutable();
+            auto peep = GetPeep();
+            if (peep != nullptr)
+            {
+                peep->hunger = value;
+            }
+        }
+
+        uint8_t thirst_get() const
+        {
+            auto peep = GetPeep();
+            return peep != nullptr ? peep->thirst : 0;
+        }
+        void thirst_set(uint8_t value)
+        {
+            ThrowIfGameStateNotMutable();
+            auto peep = GetPeep();
+            if (peep != nullptr)
+            {
+                peep->thirst = value;
+            }
+        }
+
+        uint8_t toilet_get() const
+        {
+            auto peep = GetPeep();
+            return peep != nullptr ? peep->toilet : 0;
+        }
+        void toilet_set(uint8_t value)
+        {
+            ThrowIfGameStateNotMutable();
+            auto peep = GetPeep();
+            if (peep != nullptr)
+            {
+                peep->toilet = value;
+            }
+        }
+
+        uint8_t mass_get() const
+        {
+            auto peep = GetPeep();
+            return peep != nullptr ? peep->mass : 0;
+        }
+        void mass_set(uint8_t value)
+        {
+            ThrowIfGameStateNotMutable();
+            auto peep = GetPeep();
+            if (peep != nullptr)
+            {
+                peep->mass = value;
+            }
+        }
+
+        uint8_t minIntensity_get() const
+        {
+            auto peep = GetPeep();
+            return peep != nullptr ? peep->intensity & 0x0F : 0;
+        }
+        void minIntensity_set(uint8_t value)
+        {
+            ThrowIfGameStateNotMutable();
+            auto peep = GetPeep();
+            if (peep != nullptr)
+            {
+                auto minIntensity = std::min<uint8_t>(value, 15);
+                auto maxIntensity = peep->intensity >> 4;
+                peep->intensity = (maxIntensity << 4) | minIntensity;
+            }
+        }
+
+        uint8_t maxIntensity_get() const
+        {
+            auto peep = GetPeep();
+            return peep != nullptr ? peep->intensity >> 4 : 0;
+        }
+        void maxIntensity_set(uint8_t value)
+        {
+            ThrowIfGameStateNotMutable();
+            auto peep = GetPeep();
+            if (peep != nullptr)
+            {
+                auto minIntensity = peep->intensity & 0x0F;
+                auto maxIntensity = std::min<uint8_t>(value, 15);
+                peep->intensity = (maxIntensity << 4) | minIntensity;
```

Comment:
I prefer if these implementation details were abstracted away by implementing them as methods on the `Guest` struct. The scripting layer should not have in-depth knowledge of implementation details of the Guest struct, IMO.

It is okay if you address this is another PR, but in that case: please create an issue as well.